In [1]:
import numpy as np
from scipy import stats

### Задача 1
#### Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.




Считаем распределение диаметра нормальным (отклонения обусловлены большим количеством мелких независимых случайных факторов).

- Уровень значимости $\alpha = 0.025$.

- Гипотеза H0: $\mu = 17$. 

- Cтандартное отклонение $\sigma = \sqrt{4} = 2$

Для выбранного уровня значимости  для случайной величины $T(X) = \dfrac{1}{n} \displaystyle\sum_{i = 1}^n x_i$ по правилу 2-х $\sigma$ правосторонняя критическая область будет следующей: $\left( 17 + 2 \cdot \dfrac{2}{\sqrt{100}},  \infty \right) $, т.е. $T \ge 17.4$.

### Таким образом, гипотеза отвергнута.

#### --------------------------------------------------------

### Задача 2

#### Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочное среднее равно 80, а объём равен 256.


Снова используем правило 2 $\sigma$ для нормально распределенной случайной величины $T(X) = \dfrac{1}{n} \displaystyle\sum_{i = 1}^n x_i$. 

Для доверительного интервала оценка следующая: 
### $\left( 80 - 2 \cdot \dfrac{16}{\sqrt{256}}, 80 + 2 \cdot \dfrac{16}{\sqrt{256}} \right) = (78, 82)$

#### --------------------------------------------------------

### Задача 3

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

#### Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [2]:
w = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

#### Гипотеза H0: $\mu$ = 200

In [3]:
S = w.std(ddof=1)
S

4.453463071962462

#### В предположении нормальности $w$ случайная величина ${\frac  {{\bar  {X}}- \mu }{S/{\sqrt  {n}}}}$ имеет распределение Стъюдента с $n-1$ степенью свободы

In [4]:
n = len(w)
t = stats.t(n - 1) # создадим распределение Стъюдента с n-1 степенями свободы

In [5]:
# Найдем квантиль
t.ppf(0.995) # односторонний взгляд - percent point function обратная к cdf

3.2498355440153697

In [6]:
t.interval(0.99) # то же самое в двустороннем варианте

(-3.2498355440153697, 3.2498355440153697)

#### Из $ t_{1} \le {\frac  {{\bar  {X}}- \mu }{S/{\sqrt  {n}}}} \le t_{2} \Rightarrow \mu + t_{1} \cdot $ ${\frac {S}{\sqrt  {n}} } \le \bar  {X} \le \mu + t_{2} \cdot $ ${\frac {S}{\sqrt  {n}} }$
#### Найдем этот интервал

In [7]:
(200 + t.interval(0.99)[0] * S / np.sqrt(n), 200 + t.interval(0.99)[1] * S / np.sqrt(n))

(195.42322840036354, 204.57677159963646)

In [8]:
m = w.mean() 
m

198.5

#### Итак,  $\bar  {X}=198.5 \in (195.42322840036354, 204.57677159963646)$ 

### Гипотеза H0  не может быть отвергнута